### DISCUSSION TITLE
T-Deck Private Encrypted Channels

### DISCUSSION THREAD

**Discussion by park3r1** [meshtastic/firmware#3759] [Q&A] [answered]
Hey All,

Not sure if this is the right spot to post this..new to all of this. I just got a T-Deck and cant seem to get a private channel setup or working. I want to be able to setup my own channel where my other Meshtastic devices can only communicate on. IE for like my freinds and family to be on a private channel. Any word on how to do this on the T-Deck?

**Comment by madeofstown** [comment]
The process is likely the same as most other devices. Go to the channels configuration section in the Android or IOS app and add the new channel. Then Export Settings -> select channels and save that file on your phone. Then connect to the other nodes and Import Settings -> select the file you saved earlier. Next go to your messages and send a message in the new channel.

Also, if you want to send out GPS location data on that private channel you need to set it up as your primary channel. The easiest way to accomplish this is to factory reset your device -> Then connect with your phone. DO NOT SET THE REGION! Now set up your primary [0] channel with any name you like. Reboot the node. Connect again with your phone and now you can set the region. When setting up the other nodes you should do the same factory reset process and then Import Settings for the channels before setting the region.

**Reply by park3r1** [reply]
Thanks for the response. I can do that and have it show up in the IOS app while bluetoothd to my T-Deck. When i go to send a message on the T-Deck im unable to select the Secondary channel. Only my Primary channel and a direct message to other nodes. Is this a user problem (me?) or is this something that simply hasnt been implmented yet.

**Reply by alternativesurfer** [reply, marked as answer]
The primary channel can be tabbed through to other channels/DMs.

alt+c then tap T, 1 time, allows you to tab between DMs.
alt +c, tap T, do it 2 times and you can tab between channels

**Comment by Spire9** [comment]
Same here.  I just bought 2 T-decks with the understanding my family could text each other in private if we set it up right. It seems the Meshtastic for T-deck can only send messages to the primary channel, and the primary channel refuses to be re-configured. The Config>>Security tab on each device has different private keys, but they still talk to each other, so that seems to be something different too.  Maybe I am missing something, but it feels more likely Meshtastic is missing something. :(

**Reply by park3r1** [reply]
Hey! Funny enough what I ended up doing was re-configuring the primary channel with a private key and set them on both devices. Enabling private communication, only with the other devices on the same private key. 

You should also check out the 2.6 firmware update for the T-Decks. I think it solves your problem. I'm pretty sure it enables the primary channel to talk openly(default) and the secondary to be a private channel. I think you can have up to 7 unique channels.

**Reply by Spire9** [reply]
Where did you find version 2.6? I'm running the latest version on the flasher page, 2.5.

**Reply by alternativesurfer** [reply]
The primary channel can be tabbed through to other channels/DMs.

alt+c then tap T, 1 time, allows you to tab between DMs.
alt +c, tap T, do it 2 times and you can tab between channels

**Reply by Spire9** [reply]
Thanks @alternativesurfer.  Is there a way to make it default to my DMs when the device powers on?  It seems every time I shut it off or bump the reset it needs us to press alt+c then tap T, then scroll threw 88 different nodes in no particular order,  then click the scroll wheel without rotating it.

**Reply by Spire9** [reply]
I just spent some time writing 3 sentences and pushed send. For an unknown reason the transmission failed. Where is my message? How do I retry? Do I need to write it all again? Will it fail again? How is this considered usable software???

**Reply by alternativesurfer** [reply]
Transmission failed typically means there are no nodes nearby to relay the message out.

Re usable software, consider this the same as a walkie talkie, but text. If you can't hit another radio, you can't talk to anyone.

**Reply by Spire9** [reply]
There were 2 nodes within range when I wrote it, which is typical for this spot. 
The bare minimum expected functionality would be to keep the failed message on screen.  It's just gone forever when the transmission fails.

**Reply by alternativesurfer** [reply]
iOS?

On android the messages stay present, just with diff icons based on status.

This screenshot shows all of them.
![Screenshot_20250311-174124](https://github.com/user-attachments/assets/aef2fc20-69a1-4c1a-873d-b998de2407e9)

**Comment by Spire9** [comment]
A T-deck looks like a blackberry and just has firmware, no OS. It has a full color touch screen, but the meshtastic install turns it into a 1980 looking screen and does not use the touch feature.  I will try to embed a pic for you. 
![t-deck](https://github.com/user-attachments/assets/53991a86-4291-4e82-a41e-0fa8e8bf4261)

**Reply by park3r1** [reply]
on the new 2.6 firmware is adds an OS. Makes its way more friendly.

**Comment by alternativesurfer** [comment]


**Reply by Spire9** [reply]
It seems this conversation got Schrödinger's version 2.6 added to flasher.meshtastic today. :)
